### PR TITLE
Reorder error function inputs for common error metrics

### LINKED ISSUE #380: Reorder error functions inputs y_true and y_pred
All errors functions like `accuracy_score`, `relative_error` and `absolute_error` follow the following signature convention

```Nim
proc my_error[T](y_true, y: Tensor[T]): T =
  discard
```

This is the same convention as scikit-learn, see [accuracy_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html).

However this convention hurts composition via the method call syntax.
For example, inverting the arguments would allow.
```
my_tensor.foo.bar.baz.accuracy_score(true_labels)
```
While currently we are "forced" to use function call syntax:
```
accuracy_score(true_labels, my_tensor.foo.bar.baz)
```

The potential breakage is minimal as all basic error functions are commutative. And it will not break named parameters calls.

Complex loss functions like `softmax_cross_entropy`, `sigmoid_cross_entropy` or `mean_loss` already use the (input, target) ordering.

### PR DESCRIPTION
**Pull request by berquist**
Closes #380 

I'm not sure if I caught all the function signatures, but a `rg -Fl 'y_true'` only hits these files. I also changed a few (but intentionally not most) of the examples using the error functions. Let me know if there is anything else I should change.

### CODE REVIEW

**Comment by mratsim** [comment]
Nice thanks, can you add an entry in the changelog?

**Comment by mratsim** [comment]
The examples also should be reordered, for example: https://github.com/mratsim/Arraymancer/blob/f06c43db379d1ef0160315162a45a8ab5bdb2b0d/examples/ex02_handwritten_digits_recognition.nim#L88

To:
```Nim
score += y_pred.accuracy_score(y_test[i*1000 ..< (i+1)*1000]) 
```

**Comment by berquist** [comment]
I did end of changing many instances inside tests, except for:
1. The comments for PCA and SVD
2. Cases where there was some whitespace alignment for `y_pred`:
```
        check:  mean_absolute_error( vec[_, col], expected_vec[_, col]) < 1e-4 or
                mean_absolute_error(-vec[_, col], expected_vec[_, col]) < 1e-4
```
3. Cases where:
```nim
mean_relative_error(expected_grad, expected_sparse_grad)
```

There were also a few leftovers for more complex error functions. If you don't want any of these, we can just nuke those commits.

**Comment by mratsim** [comment]
Thank you!
